app/src/main/java/com/sdp13epfl2021/projmag/database/UserProfileDatabase.kt

```diff
@@ -59,69 +65,63 @@ class UserProfileDatabase(
         }
     }
 
-    public fun getProfile(onSuccess: (profile: ImmutableProfile?) -> Unit) {
+    /**
+     * Gets the user profile from firebase if it exists
+     * @param onSuccess the closure that's when a profile is downloaded successfully with the fetched profile passed to it
+     * @param onFailure the closure that's called if the download fails
+     */
+    public fun getProfile(
+        onSuccess: (profile: ImmutableProfile?) -> Unit,
+        onFailure: (Exception) -> kotlin.Unit
+    ) {
         val userUid = getUser()?.uid
         if (userUid != null) {
```

Comment:
If the userUid is null, the code won't call the onFailure callback.

Reply:
I added the call to onFailure in an else case